Install libuser package

rpluem-vf opened the pull request:
Install libuser package on

- fedora31
- fedora32
- opensuse15
- opensuse15py2
- ubuntu1604
- ubuntu1804
- ubuntu2004

Absence of libuser caused test failures in
https://github.com/ansible/ansible/pull/71952

mattclay commented:
@rpluem-vf Why is this needed? I don't see any test failures on https://github.com/ansible/ansible/pull/71952

rpluem-vf commented:
@mattclay : Initially the tests failed with the [following](https://app.shippable.com/github/ansible/ansible/runs/174307/26/console) on Fedora:

```
Failed to find required executable lgroupmod in paths: /tmp/python-1vfuuw2z-ansible:/root/ansible/test/lib/ansible_test/_data/injector:/root/ansible/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
```
I thought that I fixed this failure by adding tags in [98057a7](https://github.com/ansible/ansible/pull/71952/commits/98057a7875be4899368d489d4a76410d1a5aceaf) which I though would not cause the tests to be executed. But the important thing was, that I moved the tests to the end (which I only discovered now when working on this reply). This caused the following [task](https://github.com/ansible/ansible/blob/6a7e637c5f5c715b148e9879fd3936d2598b77f8/test/integration/targets/user/tasks/main.yml#L996-L1003) to be executed before my tests:

```yaml
- name: Ensure luseradd is present
  action: "{{ ansible_facts.pkg_mgr }}"
  args:
    name: libuser
    state: present
  when: ansible_facts.system in ['Linux']
  tags:
    - user_test_local_mode
```

Hence `libuser` was present when my tests ran. I think the test case itself should not need to install packages in order to work like here. In the Centos 8 image `libuser`gets [installed](https://github.com/ansible/distro-test-containers/blob/7a6b4b2bba921ff47f46e5ff61539e4476b7f75f/centos8-test-container/Dockerfile#L32).

samdoran commented:
I added this package to the CentOS image originally to make it match CentOS 6 and CentOS 7 which included that package without having to be explicitly installed. That was the incorrect fix.

Rather than adding the package to each test image, the tests should be responsible for ensuring the package is installed. Since this is needed by two integration test targets, this should be put in a `setup_libuser` (or some other relevant and descriptive name) role and added as a dependency for tests that need it. The individual tasks that install `libuser` in the `group` and `user` tests should be removed.

rpluem-vf commented:
Fair enough. But it gives me a direction what approach to take in case I hit the situation in the future that a test needs a package installed.
